client/starwhale/core/dataset/type.py

```diff
@@ -702,3 +702,53 @@ def create_by_yaml(cls, fpath: t.Union[str, Path]) -> DatasetConfig:
         c = load_yaml(fpath)
 
         return cls(**c)
+
+
+class JsonDict(SwObject):
+    """
+    JsonDict takes json like dict(https://www.json.org/json-en.html) as init parameter
+    Besides the standard value types, SwObject is an extra value type that is allowed to be passed in
+    """
+
+    def __init__(self, d: dict = {}) -> None:
+        for _k, _v in d.items():
+            if type(_k) != str:
+                raise ValueError("json like dict shouldn't have none-str keys")
```

Comment:
add `_k` value into the exception message? it can help users to find out the exception reason.

Reply:
done